src/components/ArticleList.jsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect } from "react";
+import { getArticles } from "../Api";
+import ArticleCard from "./ArticleCard";
+
+const ArticleList = ({ articles, setArticles, selectedTopic }) => {
+  useEffect(() => {
+    getArticles(selectedTopic).then((data) => {
+      const filteredArticles = data.filter(
+        (article) => article.topic === selectedTopic
+      );
+      if (filteredArticles.length === 0) {
+        setArticles(data);
+      } else {
+        setArticles(filteredArticles);
+      }
+    });
+  }, [selectedTopic]);
+
+  return <ArticleCard articles={articles} />;
```

Comment:
instead of the ArticleCard component returning here it would be better to put your map here and have this ArticleCard the return value of that map